akka-stream/src/main/scala/akka/stream/impl/Timers.scala

```diff
@@ -100,15 +106,43 @@ private[stream] object Timers {
     override def toString = "IdleTimeout"
   }
 
+  final class BackpressureTimeout[T](timeout: FiniteDuration) extends SimpleLinearGraphStage[T] {
+    override def initialAttributes = DefaultAttributes.backpressureTimeout
+
+    override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new TimerGraphStageLogic(shape) {
+      private var nextDeadline: Option[Deadline] = None
+
+      setHandler(in, new InHandler {
+        override def onPush(): Unit = {
+          push(out, grab(in))
+          nextDeadline = Some(Deadline.now + timeout)
+        }
+      })
+
+      setHandler(out, new OutHandler {
+        override def onPull(): Unit = {
+          nextDeadline = None
+          pull(in)
+        }
+      })
+
+      final override protected def onTimer(key: Any): Unit =
+        nextDeadline.foreach(
+          e ⇒ if (e.isOverdue()) failStage(new TimeoutException(s"No backpressure notified for last $timeout.")))
+
+      override def preStart(): Unit = schedulePeriodically("BackpressureTimeoutCheckTimer", interval = idleTimeoutCheckInterval(timeout))
```

Comment:
use a `case object BackpressureTimeoutCheckTimer` defined in the `BackpressureTimeout` companion instead of a string